dev/Microsoft.Process.Environment/EnvironmentManager.cpp

```diff
@@ -141,12 +136,147 @@ namespace winrt::Microsoft::ProjectReunion::implementation
 
     void EnvironmentManager::AppendToPath(hstring const& path)
     {
-        throw hresult_not_implemented();
+        // Get the existing path because we will append to it.
+        std::wstring existingPath = GetPath();
+
+        // Don't append to the path if the addition already exists.
+        if (existingPath.find(path) != std::wstring::npos)
+        {
+            return;
+        }
+
+        std::wstring newPath = existingPath.append(path);
+
+        if (newPath.back() != L';')
+        {
+            newPath += L';';
+        }
+
+        auto setPath = [&, newPath, this]()
+        {
+            if (m_Scope == Scope::Process)
+            {
+                BOOL result = FALSE;
+                result = ::SetEnvironmentVariable(L"Path", newPath.c_str());
+
+                if (result == 0)
+                {
+                    THROW_HR(HRESULT_FROM_WIN32(GetLastError()));
+                }
+            }
+            else //Scope is either user or machine
+            {
+                wil::unique_hkey environmentVariableKey = GetRegHKeyForEVUserAndMachineScope(true);
+
+                THROW_IF_FAILED(HRESULT_FROM_WIN32(RegSetValueEx(
+                    environmentVariableKey.get()
+                    , L"Path"
+                    , 0
+                    , REG_EXPAND_SZ
+                    , reinterpret_cast<const BYTE*>(newPath.c_str())
+                    , static_cast<DWORD>((newPath.size() + 1) * sizeof(wchar_t)))));
+
+                LRESULT broadcastResult = SendMessageTimeout(HWND_BROADCAST, WM_SETTINGCHANGE,
+                    reinterpret_cast<WPARAM>(nullptr), reinterpret_cast<LPARAM>(L"Environment"),
+                    SMTO_NOTIMEOUTIFNOTHUNG | SMTO_BLOCK, 1000, nullptr);
+
+                if (broadcastResult == 0)
+                {
+                    THROW_HR(HRESULT_FROM_WIN32(GetLastError()));
+                }
+            }
+
+            return S_OK;
+        };
+
+        EnvironmentVariableChangeTracker changeTracker(L"Path", std::wstring(newPath), m_Scope);
+
+        THROW_IF_FAILED(changeTracker.TrackChange(setPath));
     }
 
     void EnvironmentManager::RemoveFromPath(hstring const& path)
     {
-        throw hresult_not_implemented();
+        // A user is only allowed to remove something from the PATH if
+        // 1. path exists in PATH
+        // 2. path matches a path part exactly (ignoring case)
+        std::wstring currentPath = GetPath();
+        std::wstring pathPartToFind(path);
+
+        if (pathPartToFind.back() != L';')
```

Comment:
This throws if `RemoveFromPath` is called with an empty string.

Reply:
Nice catch.  Fixed.